Tasks/MavenAuthenticateV0/mavenauth.ts

```diff
@@ -13,12 +15,16 @@ tl.setResourcePath(path.join(__dirname, 'task.json'));
 async function run(): Promise<void> {
     let internalFeedServerElements: any[] = [];
     let externalServiceEndpointsServerElements: any[] = [];
+    let federatedFeedAuthSuccessCount: number = 0;
     try {
+        const feedUrl = tl.getInput("feedUrl");
+        const entraWifServiceConnectionName = tl.getInput("workloadIdentityServiceConnection");
+
         internalFeedServerElements = util.getInternalFeedsServerElements("artifactsFeeds");
         externalServiceEndpointsServerElements = util.getExternalServiceEndpointsServerElements("mavenServiceConnections");
         const newServerElements = internalFeedServerElements.concat(externalServiceEndpointsServerElements);
 
-        if(newServerElements.length === 0) {
+        if(newServerElements.length === 0 && (!feedUrl || !entraWifServiceConnectionName)) {
```

Comment:
I think with maven we would still want to use a lot of the code that comes after this check because it creates or copies an existing settings.xml file. For now it's also possible to use both a WIF connection and regular feeds and maven service connections in the same run. We can change this, but it seemed like more work to just allow one at a time, and I don't think allowing both makes it confusing

Reply:
Changed this so that only the new service connection or internal/ external feeds can be authenticated in a single run